test/integration/puppetlabs/services/jruby/jruby_pool_int_test.clj

```diff
@@ -166,6 +198,42 @@
         ;; now the pool is flushed, and the constants should be cleared
         (is (true? (verify-no-constants pool-context 4)))))))
 
+(deftest ^:integration hold-file-handle-on-instance-while-pool-flush-in-progress-test
+  (testing "file handle opened from old pool instance is held open across pool flush"
+    (bootstrap/with-puppetserver-running
+      app
+      {:puppet-admin {:client-whitelist ["localhost"]}
+       :jruby-puppet {:max-active-instances 4}}
+      (let [jruby-service (tk-app/get-service app :JRubyPuppetService)
+            context (tk-services/service-context jruby-service)
+            pool-context (:pool-context context)]
+        ;; set a ruby constant in each instance so that we can recognize them
+        (is (true? (set-constants-and-verify pool-context 4)))
+        (let [flush-complete (add-watch-for-flush-complete pool-context)
+              ;; borrow an instance and hold the reference to it.
+              instance (jruby-protocol/borrow-instance jruby-service)
+              sc (:scripting-container instance)]
+          (.runScriptlet sc
+                         (str "$unique_file = "
+                              "Puppet::FileSystem::Uniquefile.new"
+                              "('hold-instance-test-', './target')"))
+          (try
+            ;; trigger a flush
+            (is (true? (trigger-flush ssl-request-options)))
+            ;; wait for the new pool to become available
+            (wait-for-new-pool jruby-service)
+
+            (is (nil? (.runScriptlet sc "$unique_file.close"))
```

Comment:
This line is basically what would cause the IOException / BadFileDescriptor errors that we were seeing back before SERVER-38 was fixed, #227.  A nil return here is what we want - indicating that JRuby hasn't unintentionally stomped the file descriptor out from underneath the container as a result of any terminates being called on other `ScriptingContainer`s.